src/autoscaler/scalingengine/server/applock_test.go

```diff
@@ -0,0 +1,106 @@
+package server_test
+
+import (
+	. "autoscaler/scalingengine/server"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"time"
+)
+
+var (
+	ch      chan string
+	appLock *AppLock
+	err     interface{}
+)
+
+var _ = Describe("Applock", func() {
+	BeforeEach(func() {
+		appLock = NewAppLock()
```

Comment:
for 1)   Do we have concerns of holding too much memory, only one lock is created per app? how do we decide when to remove the Mutex from map? 
For 2)  solved by passing AppLock as an argument. I did not realize the data race is across test cases.